Src/Base/AMReX_Random.H

```diff
@@ -99,8 +121,13 @@ namespace amrex
         mkl::rng::device::poisson<unsigned int> distr(lambda);
         return mkl::rng::device::generate(distr, *random_engine.engine);
 #else
-        amrex::ignore_unused(random_engine);
-        return RandomPoisson(lambda);
+        AMREX_IF_ON_DEVICE((
```

Comment:
This is also similar. Our assumption is only cuda device code can get here and has a non-empty AMREX_IF_ON_DEVICE. So we don't really need the `if defined __CUDA_ARCH__`. In fact, we can probably remove `__CUDA_ARCH__` entirely from amrex, except where we define AMREX_IF_ON_DEVIOCE.

Reply:
Fixed. Also fixed in AMReX_Math.H.

Remaining uses of `__CUDA_ARCH__` are:
- AMReX_Extension.H (used to conditionally define `AMREX_PRAGMA_SIMD` and `AMREX_FORCE_INLINE`)
- AMReX_GpuAtomic.H (one use of `#if defined(AMREX_USE_CUDA) && (__CUDA_ARCH__ < 600)`)
- AMReX_Parser_Y.H (used for elliptic integrals that are only supported on host GCC)

Reply:
What do you want to do about the `__CUDA_ARCH__` in AMReX_Extension.H? I don't think the latter two files really matter.

Reply:
The use of `__CUDA_ARCH__` in AMReX_Extenstion.H is fine.

In AMReX_GpuAtomic.H, I think we need to change from `#if defined(AMREX_USE_CUDA) && (__CUDA_ARCH__ < 600)` to `#if defined(AMREX_USE_CUDA) && defined(__CUDA_ARCH__) && (__CUDA_ARCH__ < 600)`, because `__CUDA_ARCH__` would be become 0 in the comparison if it's not define.

As for AMReX_Parser_Y.H, `nvc++ -E -dM - < /dev/null` shows `#if defined(__GNUC__) && !defined(__clang__) && !defined(__CUDA_ARCH__)` is true. So we need to update it unless `std::comp_ellint_1` and `_2` work on device with nvhpc .